docs/architecture/overview.md

```diff
@@ -1,4 +1,116 @@
 # Overview
 
-!!! warning ""
-    Under construction.
+Hedgehog Open Network Fabric is built leverages the Kubernetes API to manage its resources. All user-facing APIs are exposed as Kubernetes Custom Resources Definitions (CRDs), allowing users to manage Fabric resources using standard Kubernetes tools.
+
+Since network switches are not inherently Kubernetes-aware, Hedgehog employs a **Fabric Agent** running on each switch. This agent acts as a bridge between the Kubernetes control plane and the switch’s internal network configuration mechanisms. It continuously syncs desired state from Kubernetes via the Fabric Controller and applies configurations using **gNMI** (gRPC Network Management Interface).
+
+## Components
+
+Hedgehog Fabric consists of several key components, distributed between the Control Node and the Network devices. The following diagram decomposes a [Collapsed Core](https://docs.githedgehog.com/latest/architecture/fabric/#collapsed-core) Topology. Hedgehog components have been highlighted in sandy brown color:
+
+``` mermaid
+graph TD;
+  %% Control Plane
+  subgraph Control Node ["Control Node"]
+    K[Fabric CLI - kubectl plugin]:::ourComponent
+    A[Fabric Controller]:::ourComponent
+    P[Fabric Proxy]:::ourComponent
+  end
+
+  K -->|Interacts via K8s API| A
+  L[Fabricator]:::ourComponent -->|Installs & Configures| A
+  A -->|Kubernetes API| B1
+  B1 -->|Syncs State| A;
+  A -->|Kubernetes API| B2
+  B2 -->|Syncs State| A;
+
+  %% Collapsed Core - Two Switches
+  subgraph SONiC Leaf 2
+    B1[Fabric Agent]:::ourComponent -->|Scraped by| C1[Alloy]
+    C1 -->|Logs/Metrics| P
+    D1[gNMI]
+    E1[Config DB]
+    I1[ASIC]
+  end
+
+  subgraph SONiC Leaf 1
+    B2[Fabric Agent]:::ourComponent -->|Scraped by| C2[Alloy]
+    C2 -->|Logs/Metrics| P
```

Comment:
When describing metrics, I think it would help put people at ease if we mention that we are **pushing** the metrics via alloy. That way people know that grafana cloud or a local instance isn't going to be reaching into the switches. So for the label of  `Logs/ Metrics` I would change to `Pushing Logs/Metrics`.